node_modules/oae-jitsi/lib/activity.js

```diff
@@ -0,0 +1,553 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('underscore');
+
+var ActivityAPI = require('oae-activity');
+var ActivityConstants = require('oae-activity/lib/constants').ActivityConstants;
+var ActivityModel = require('oae-activity/lib/model');
+var ActivityUtil = require('oae-activity/lib/util');
+var AuthzAPI = require('oae-authz');
+var AuthzConstants = require('oae-authz/lib/constants').AuthzConstants;
+var AuthzUtil = require('oae-authz/lib/util');
+var Context = require('oae-context').Context;
+var log = require('oae-logger').logger('oae-bbb');
+var MessageBoxAPI = require('oae-messagebox');
+var MessageBoxUtil = require('oae-messagebox/lib/util');
+var PrincipalsDAO = require('oae-principals/lib/internal/dao');
+var PrincipalsUtil = require('oae-principals/lib/util');
+var Tenant = require('oae-tenants/lib/model').Tenant;
+var TenantsAPI = require('oae-tenants');
+var TenantsUtil = require('oae-tenants/lib/util');
+var User = require('oae-principals/lib/model').User;
+
+var MeetingsAPI = require('./api');
+var MeetingsConstants = require('./constants').MeetingsConstants;
+var MeetingsDAO = require('./internal/dao');
+
+////////////////////
+// MEETING-CREATE //
+////////////////////
+
+ActivityAPI.registerActivityType(MeetingsConstants.activity.ACTIVITY_MEETING_CREATE, {
+    'groupBy': [{'actor': true}],
+    'streams': {
+        'activity': {
+            'router': {
+                'actor': ['self', 'followers'],
+                'object': ['self', 'members']
+            }
+        },
+        'notification': {
+            'router': {
+                'object': ['members']
+            }
+        },
+        'email': {
+            'router': {
+                'object': ['members']
+            }
+        }
+    }
+});
+
+/*!
+ * Post a meeting-create activity when a user creates a meeting.
+ */
+MeetingsAPI.on(MeetingsConstants.events.CREATED_MEETING, function(ctx, meeting, memberChangeInfo) {
+
+    var millis = Date.now();
+    var actorResource = new ActivityModel.ActivitySeedResource('user', ctx.user().id, {'user': ctx.user()});
+    var objectResource = new ActivityModel.ActivitySeedResource('meeting-jitsi', meeting.id, {'meeting-jitsi': meeting});
+    var targetResource = null;
+
+    // Get the extra members
+    var extraMembers = _.chain(memberChangeInfo.changes)
+        .keys()
+        .filter(function (member) {
+            return (member !== ctx.user().id)
+        })
+        .value();
+
+    // If we only added 1 extra user or group, we set the target to that entity
+    if (extraMembers.length === 1) {
+        var targetResourceType = (PrincipalsUtil.isGroup(extraMembers[0])) ? 'group' : 'user';
```

Comment:
Since we're getting the `extraMembers[0]` twice, would consider extracting it into a variable (also maybe using `_.first` instead).